app/src/main/java/com/example/streetworkout/data/resource/remote/util/RetrofitClient.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.streetworkout.data.resource.remote.util
+
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+
+class RetrofitClient {
+    companion object {
+        private var instance: Retrofit? = null
+        fun getInstance(url: String): Retrofit =
+            instance ?: Retrofit.Builder()
+                .baseUrl(url)
+                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+                .addConverterFactory(GsonConverterFactory.create())
+                .build().also { instance = it }
+    }
+}
```

Comment:
Sao lại truyền url vào nhỉ, dùng nhiều api à ông
```suggestion
object ApiFactory {
        fun getInstance(url: String): Retrofit =
            instance ?: Retrofit.Builder()
                .baseUrl(url)
                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create())
                .build()
    }
}
```

Reply:
Có 1 api về food nữa t đang suy nghĩ xem có lấy, t định tự viết vì api kia k ổn cho lắm